iset.mm

```diff
@@ -147216,6 +147339,173 @@ since the target of the homomorphism (operator ` O ` in our model) need
+        UNUSPBTDQRDUQUPHOBSUAKUPUBPUPUCGUDUEUFUIUMCURHBUQCDEUQUGUHUJUKUL $.
+    $}
+
+    ${
+      mgpsca.s $e |- S = ( Scalar ` R ) $.
+      $( The multiplication monoid has the same (if any) scalars as the
+         original ring.  (Contributed by Mario Carneiro, 12-Mar-2015.)
+         (Revised by Mario Carneiro, 5-May-2015.) $)
+      mgpscag $p |- ( R e. V -> S = ( Scalar ` M ) ) $=
+        ( wcel csca cfv cnx cplusg cop csts co cvv wceq mulrslid slotex scaslid
+        cmulr scandxnplusgndx cslot cn plusgslid simpri setsslnid mpdan mgpvalg
+        eqid fveq2d eqtr4d eqtrid ) ADGZBAHIZCHIZFUMUNAJKIZATIZLMNZHIZUOUMUQOGU
+        NUSPATDQRDUQUPHOASUAKUPUBPUPUCGUDUEUFUGUMCURHAUQCDEUQUIUHUJUKUL $.
+    $}
+
+    $( Topology component of the multiplication group.  (Contributed by Mario
+       Carneiro, 5-Oct-2015.) $)
+    mgptsetg $p |- ( R e. V -> ( TopSet ` R ) = ( TopSet ` M ) ) $=
+      ( wcel cts cfv cnx cplusg cmulr cop csts co wceq mulrslid slotex tsetslid
+      cvv tsetndxnplusgndx cslot plusgslid simpri setsslnid eqid mgpvalg fveq2d
+      cn mpdan eqtr4d ) ACEZAFGZAHIGZAJGZKLMZFGZBFGUJUMREUKUONAJCOPCUMULFRAQSIU
+      LTNULUGEUAUBUCUHUJBUNFAUMBCDUMUDUEUFUI $.
+
+    ${
+      mgptopn.2 $e |- J = ( TopOpen ` R ) $.
+      $( Topology of the multiplication group.  (Contributed by Mario Carneiro,
+         5-Oct-2015.) $)
+      mgptopng $p |- ( R e. V -> J = ( TopOpen ` M ) ) $=
+        ( wcel cts cfv cbs crest co ctopn topnvalg mgptsetg mgpbasg oveq12d cvv
+        eqid cmgp eqtr3d eqtrid wceq fnmgp elex funfvex funfni sylancr eqeltrid
+        wfn syl eqtrd ) ADGZBCHIZCJIZKLZCMIZUMBAMIZUPFUMAHIZAJIZKLURUPUTUSDAUTS
+        ZUSSNUMUSUNUTUOKACDEOUTACDEVAPQUAUBUMCRGUPUQUCUMCATIZREUMTRUJARGVBRGZUD
+        ADUEVCRATATUFUGUHUIUOUNRCUOSUNSNUKUL $.
+    $}
+
+    ${
+      mgpds.2 $e |- B = ( dist ` R ) $.
+      $( Distance function of the multiplication group.  (Contributed by Mario
+         Carneiro, 5-Oct-2015.) $)
+      mgpdsg $p |- ( R e. V -> B = ( dist ` M ) ) $=
+        ( wcel cds cfv cnx cplusg cmulr cop csts co wceq mulrslid slotex dsslid
+        cvv dsndxnplusgndx cslot plusgslid simpri setsslnid eqid mgpvalg fveq2d
+        cn mpdan eqtr4d eqtrid ) BDGZABHIZCHIZFUMUNBJKIZBLIZMNOZHIZUOUMUQTGUNUS
+        PBLDQRDUQUPHTBSUAKUPUBPUPUIGUCUDUEUJUMCURHBUQCDEUQUFUGUHUKUL $.
+    $}
+  $}
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Ring unit
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c 1r $.
+
+  $( Extend class notation with ring unit. $)
+  cur $a class 1r $.
+
+  $( Define the multiplicative neutral element of a ring.  This definition
+     works by extracting the ` 0g ` element, i.e. the neutral element in a
+     group or monoid, and transferring it to the multiplicative monoid via the
+     ` mulGrp ` function ( ~ df-mgp ).  See also ~ dfur2g , which derives the
```

Comment:
This is hard to write in words. Let's try:

> This definition works by transferring the multiplicative operation from the ` .r ` slot to the ` +g ` slot (via ` mulgrp ` , ~ df-mgp ) and then looking at the element which is now the ` 0g ` element, that is an identity with respect to the operation which started out in the ` .r ` slot.

Reply:
> This is hard to write in words. Let's try:
> 
> > This definition works by transferring the multiplicative operation from the `.r` slot to the `+g` slot (via `mulgrp` , ~ df-mgp ) and then looking at the element which is now the `0g` element, that is an identity with respect to the operation which started out in the `.r` slot.

In my opinion, this description seems to be too technical and complicated (although it is absolutely correct). The ring unit is simply the neutral element of the multiplicative monoid of the ring.

Maybe others have other ideas/suggestions.

Reply:
> In my opinion, this description seems to be too technical and complicated (although it is absolutely correct). The ring unit is simply the neutral element of the multiplicative monoid of the ring.

I guess it depends on one's mathematical background. To me, operations and slots are pretty easy, but "multiplicative monoid" takes more thought to get my head around. I'm also not sure I'd use the word "ring" as much in this section as I would later on (for several reasons, including that we haven't defined ring until a few sections later). (There's also "identity element" versus "neutral element" but since those are synonyms I'm not sure I have a strong reason to pick one or the other except that "identity" is more familiar to me).

> Maybe others have other ideas/suggestions.

Sounds good to me, although if we don't hear anything we should pick something (one of the proposals here, or just leave the existing wording in place) for purposes of this pull request. We can always revise it again later if desired.

Reply:
```
     Define the multiplicative identity, i.e., the group identity ( ~ df-0g )
     of the multiplicative group ( ~ df-mgp ).
```
perhaps the two sentences could be combined, I don't really see how the longer sentence explains anything more than this suggestion. Not sure if the "group" of "group identity" is necessary.

Reply:
> perhaps the two sentences could be combined, 

Yes, this is a very good suggestion

> Edit: alternative
> 
> ```
>      Define the multiplicative identity, i.e., the monoid identity ( ~ df-0g )
>      of the multiplicative monoid ( ~ df-mgp ).
> ```

Although not defined yet, I would add/keep the word "ring", because for a meaningful multiplicative monoid, a second operation ("ring multiplication") is necessary. Furthermore, the comment of df-mpg also talks about "rings"). Maybe "ring-like structure" is more general.
 
So we could have

`Define the multiplicative identity, i.e., the monoid identity ( ~ df-0g ) of the multiplicative monoid ( ~ df-mgp ) of a ring-like structure. This definition works by transferring the multiplicative operation from the .r slot to the +g slot (via mulgrp , ~ df-mgp ) and then looking at the element which is now the 0g element, that is an identity with respect to the operation which started out in the .r slot.`

Reply:
I have pushed this wording (with small tweaks) to this branch.